functional-samples/tutorial.terminate-sw/puppeteer/index.test.js

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// eslint-disable-next-line no-undef
+const puppeteer = require('puppeteer');
+
+const EXTENSION_PATH = '../test-extension';
+const EXTENSION_ID = 'gjgkofgpcmpfpggbgjgdfaaifcmoklbl';
+
+let browser;
+
+beforeEach(async () => {
+  browser = await puppeteer.launch({
```

Comment:
Is it required to relaunch the browser after each test?

Reply:
It's not a requirement but generally considered good practice to keep tests isolated. I could go either way but merging for now to time with the documentation.